File: Dished/Layout/DAReplaceSegue.m
//
//  DAReplaceSegue.m
//  Dished
//
//  Created by Ryan Khalili on 6/7/14.
//  Copyright (c) 2014 Dished. All rights reserved.
//

#import "DAReplaceSegue.h"

@implementation DAReplaceSegue

- (void)perform
{
    UIViewController *source = self.sourceViewController;
    UIViewController *dest   = self.destinationViewController;
    UINavigationController *navigationController = source.navigationController;
    
    [navigationController popToRootViewControllerAnimated:NO];
    [navigationController pushViewController:dest animated:YES];
}

@end

File: Dished/Layout/DAReplaceSegue.swift
//
//  DAReplaceSegue.swift
//  Dished
//
//  Created by Ryan Khalili on 9/22/14.
//  Copyright (c) 2014 Dished. All rights reserved.
//

import UIKit

class DAReplaceSegue: UIStoryboardSegue
{
    override func perform()
    {
        let source = self.sourceViewController as! UIViewController
        let dest = self.destinationViewController as! UIViewController
        let navigationController = source.navigationController
        
        let controllerStack = navigationController?.viewControllers
        var stackCopy = NSMutableArray( array: controllerStack! )
        stackCopy.replaceObjectAtIndex( stackCopy.indexOfObject( source ), withObject: dest )
        navigationController?.setViewControllers( stackCopy as [AnyObject], animated: true )
    }
}